MetaMask/metamask-mobile
# chore: persist fix #8550

## **Description**

Cherry-pick fix for release blocker and redux persist bug.

## **Related issues**

Fixes: #8550 

## **Manual testing steps**

Scenario 1. Upgrade 7.14.0 no vault prompt
Scenario 2. Upgrade 7.15.0 no vault prompt
Scenario 3. 
- Add network/token/contact/nft 
- kill the app, 
- start and see network/token/contact/nft, 
- remove network/token/contact/nft
- Removed values are still gone

Scenario 4. [Run nightly performance automation test ](https://app.bitrise.io/build/7179d286-bb3b-4c66-aa3e-90471fe074ae)

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
